### PR TITLE
Add github changelog generation

### PR DESCRIPTION
**Pull request by toabctl**
There are now 2 new parameters, "changelog-provider" and
"changelog-email". When both are given and a .changes file
is already there, this file will be updated with the changes detected.

### CODE REVIEW

File: renderspec [hunk 1]
```diff
@@ -18,13 +18,100 @@ from __future__ import print_function
...
+GITHUB_CHANGES = ("https://api.github.com/repos/%(owner)s/%(repo)s/"
+                  "compare/%(v1)s...%(v2)s")
+
+
+def _get_changelog_github(owner, repo, v1, v2):
+    """get a list with changes for the given versions"""
+    url = GITHUB_CHANGES % dict(owner=owner, repo=repo,
+                                v1=v1, v2=v2)
+    try:
+        res = urllib.urlopen(url)
```
Comment (dirkmueller) [inline renderspec]: Why urllib when it already depends on requests?

File: renderspec [hunk 2]
```diff
@@ -18,13 +18,100 @@ from __future__ import print_function
...
+    except IOError as exc:
+        sys.exit(exc)
```
Comment (dirkmueller) [inline renderspec]: Useful error message for Jenkins runs missing here

File: renderspec [hunk 3]
```diff
@@ -18,13 +18,100 @@ from __future__ import print_function
...
+        else:
+            # expect another list - indent2
+            for cc in c:
+                changes_str += '  - %s\n' % (cc)
+    # footer
+    changes_str += '\n'
+    return changes_str
+
+
+def _prepend_str_to_file(filename, string):
```
Comment (dirkmueller) [inline renderspec]: Prepend_string.. ? Matching argument  naming and function naming

File: renderspec [hunk 4]
```diff
@@ -92,16 +184,35 @@ if __name__ == '__main__':
...
+        # get the version *after* we run renderspec
+        new_spec_version = _get_spec_version(outfile)
+
+        # check for changelog generation
+        changes_file = outfile.replace('.spec', '.changes')
+        if old_spec_version != new_spec_version and \
+           os.path.exists(changes_file):
+            print("version changed: '%s' -> '%s'"
+                  % (old_spec_version, new_spec_version))
+            changes = ['update to version %s:' % new_spec_version]
```
Comment (dirkmueller) [inline renderspec:210]: Uppercase Update

File: renderspec (hunk 1 above)
Comment (toabctl) [reply renderspec]: done

File: renderspec (hunk 2 above)
Comment (toabctl) [reply renderspec]: done

File: renderspec (hunk 3 above)
Comment (toabctl) [reply renderspec]: done

File: renderspec (hunk 4 above)
Comment (toabctl) [reply renderspec:210]: done

**Comment by dirkmueller** [comment]
extremely cool, thanks!

**Pull request merged**